File: win95OEM.py
__author__ = "ThatScode"

import random

def trial():
	d1 = random.randint(100, 366)
	d2 = random.randint(0, 366)
	d3 = random.randint(0, 366)
	d4 = random.randint(95, 99)
	d5 = 0
	d6 = random.randint(0, 9)
	d7 = random.randint(0, 9)
	d8 = random.randint(0, 9)
	d9 = random.randint(0, 9)
	d10 = random.randint(0, 9)
	d11 = random.randint(0, 9)
	d12 = random.randint(0, 9)
	d13 = random.randint(0, 9)
	d14 = random.randint(0, 9)
	d15 = random.randint(0, 9)
	d16 = random.randint(0, 9)
	
	z = int(str(d5) + str(d6) + str(d7) + str(d8) + str(d9) + str(d10) + str(d11))
	

	if int(z) % 7 == 0:
		if not d16 == 0 and not d16 == 8 and not d16 == 9:
			print(f"{d1}{d4}-OEM-{d5}{d6}{d7}{d8}{d9}{d10}{d11}-{d12}{d13}{d14}{d15}{d16}")


		else:
			A = 1

	else:
		a = 1

numtrials = int(input(f"Program made by {__author__}. How many trials do you want? : "))

for i in range(numtrials):
	trial()

File: win95OEM.js
const author = 'miha_focsa';

let licenses = 0;

let genFirstNumber = () => {
    let number = Math.floor((Math.random() * 366 ) + 1);

    if(number <= 9) {
        number = `0${number}`;
    }
    if(number <= 99) {
        number = `0${number}`;
    }

    return number;
}
let genSecondNumber = () => {
    let number = Math.floor((Math.random() * 92 ) + 3);
    
    if(number <= 9) {
        number = `00${number}`;
    } else {
        number = `0${number}`;
    }

   return number;
}
let trial = () => {
    
    let year = genFirstNumber();
    let day = genSecondNumber();
    let d0 = 0;
    let d6 = Math.floor(Math.random() * 9 );
	let d7 = Math.floor(Math.random() * 9 );
	let d8 = Math.floor(Math.random() * 9 );
	let d9 = Math.floor(Math.random() * 9 );
	let d10 = Math.floor(Math.random() * 9 );
	let d11 = Math.floor(Math.random() * 9 );
    
    
    let d12 = Math.floor(Math.random() * 9 );
    let d13 = Math.floor(Math.random() * 9 );
    let d14 = Math.floor(Math.random() * 9 );
    let d15 = Math.floor(Math.random() * 9 );
    let d16 = Math.floor(Math.random() * 9 );
    
    let z = `${d0}${d6}${d7}${d8}${d9}${d10}${d11}`;

    if(z % 7 == 0) {
        console.log(`${year}${day}-OEM-${d0}${d6}${d7}${d8}${d9}${d10}${d11}-${d12}${d13}${d14}${d15}${d16}`);
        licenses++;
    }
}

let wantedLicenses = window.prompt(`Program made by ${author}. How many licenses do you want? : `);

while(wantedLicenses != licenses) {
    trial();
}


